File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int n = s.length();
        string ans = "";
        for(int i=0;i<n;i++){
            if((s[i]>='0' && s[i]<='9') || (s[i]>='A' && s[i]<='Z') || (s[i]>='a' && s[i]<='z')) {
                ans += tolower(s[i]);
            }
        }
        string a = ans;
        reverse(ans.begin(),ans.end());
        return ans==a;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int n = s.length();
        String ans = "";
        for(int i=0;i<n;i++){
            char ch = s.charAt(i);
            if(ch >= 'A' && ch <='Z'){
                ch = (char)(ch + 32);
                ans += ch;
            }
            else if(ch >= 'a' && ch <='z'){
                ans += ch;
            }
            else if(ch >= '0' && ch <= '9'){
                ans += ch;
            }
        }
        int i=0;
        int c = ans.length();
        int j = c-1;
        while(i<j){
            if(ans.charAt(i)==ans.charAt(j)){
                i++;
                j--;
            }
            else return false;
        }
        return true;
    }
}